### PR TITLE
Update `generate-regression-tests.sh` work with modern evm-semantics

### LINKED ISSUE #3686: Update `generate-regression-tests.sh` to work with current evm-semantics
The script `generate-regression-tests.sh` used to re-generate all of the tests in `test/regression-evm` is out of date and no longer works with recent versions of the evm-semantics. 

Specifically, it relies on the old evm-semantics Makefile-based build system, and needs to be updated to use the `poetry -C kevm-pyk ...` based commands instead.

### PR DESCRIPTION
**Pull request by Scott-Guest**
Fixes #3686

This PR does the following
- Updates `generate-regression-tests.sh` to work with modern versions of evm-semantics which no longer use a Makefile-based build system.
- Fixes all issues reported by `shellcheck`.
- Runs the new script and updates all the `regression-evm` tests.

---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [ ] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [ ] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [ ] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [ ] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

### CODE REVIEW

File: scripts/generate-regression-tests.sh
Comment (Scott-Guest) [inline scripts/generate-regression-tests.sh:137]: The `kevm-bug-*.tar.gz` glob here is to work around some weird behavior where `functional/lemmas` generates a bug report named `kevm-bug-lemma.tar.gz` (without the "s"). I updated the script to run each test in its own directory, so the glob should find a unique file.

File: test/regression-evm/test-totalSupply.sh
Comment (Scott-Guest) [inline test/regression-evm/test-totalSupply.sh:13]: Ah need to `sed` the script as well, will address this

File: test/regression-evm/test-dsvalue-peek-pass-rough.sh
Comment (jberthold) [inline test/regression-evm/test-dsvalue-peek-pass-rough.sh:13]: :raised_eyebrow:

File: test/regression-evm/test-totalSupply.sh
Comment (jberthold) [reply test/regression-evm/test-totalSupply.sh:13]: Why are we even producing these logs? This should probably be removed from the `evm-semantics` code.
But yes, a `sed` script to kill the `--log` and its argument (either one or two lines, depending on `\` being present)  is good enough. Or I'd suggest to rather  just fix this manually now, we are not running this  script automatically and the correct fix is to change `evm-semantics`.

**Review by jberthold — APPROVED** [review]
Pre-approving, provided the scripts are changed (manually) before merging

**Comment by Scott-Guest** [comment]
@jberthold It looks like Pyk (and correspondingly KEVM) always enables logging in order to detect whether a proof took zero steps: https://github.com/runtimeverification/pyk/blob/fd2cd65b01919ac608e0c4f8a8d21251f5d8a973/src/pyk/ktool/kprove.py#L252

I ended up just sed-ding out the `--log` entry rather than removing this check.

**Pull request merged**
